src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,406 @@
+        {
+            // If we're in the Function root, the .csproj file will be in this folder
+            var csProjFiles = FileSystemHelpers.GetFiles(Environment.CurrentDirectory, searchPattern: "*.csproj").ToList();
+            if (csProjFiles.Count == 1)
+            {
+                return csProjFiles.First();
+            }
+
+            // If we're in the function root\bin\debug\netstandard2.x, the .csproj file will be up three directories
+            try
+            {
+                var functionDir = Path.GetFullPath(Path.Combine(Environment.CurrentDirectory, @"..\..\..\"));
+                var functionDirProjFiles = FileSystemHelpers.GetFiles(functionDir, searchPattern: "*.csproj").ToList();
+
+                if (functionDirProjFiles.Count == 1)
+                {
+                    return functionDirProjFiles.First();
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            catch (DirectoryNotFoundException e)
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Modify the Function's .csproj so the middleware auth json file will copy to the output directory
+        /// </summary>
+        private static void ModifyCSProj(string csProj)
+        {
+            var xmlFile = XDocument.Load(csProj);
+
+            var project = xmlFile.Element("Project");
+            var itemGroups = project.Elements("ItemGroup");
+
+            var existing = itemGroups.Elements("None").FirstOrDefault(elm => elm.Attribute("Update").Value.Equals(SecretsManager.MiddlewareAuthSettingsFileName));
+            if (existing != null)
+            {
+                // If we've added this file to the .csproj during a previous create-aad call, do not add again
+                return;
+            }
+
+            // Assemble the attribute
+            var newItemGroup = new XElement("ItemGroup");
+            var noneElement = new XElement("None", new XAttribute("Update", SecretsManager.MiddlewareAuthSettingsFileName));
+            noneElement.Add(new XElement("CopyToOutputDirectory", "PreserveNewest"));
+            noneElement.Add(new XElement("CopyToPublishDirectory", "Never"));
+            newItemGroup.Add(noneElement);
+
+            // Append item group to project & save
+            project.Add(newItemGroup);
+            xmlFile.Save(csProj);
+
+            ColoredConsole.WriteLine(Yellow($"Modified {csProj} to include {SecretsManager.MiddlewareAuthSettingsFileName} in output directory."));
+        }
+
+        private Dictionary<string, string> CreateAuthSettingsToPublish(string clientId, string clientSecret, string tenant, List<string> replyUrls)
+        {
+            // The 'allowedAudiences' setting of /config/authsettings is of the form ["{replyURL1}", "{replyURL2}"]
+            string serializedArray = JsonConvert.SerializeObject(replyUrls, Formatting.Indented);
```

Comment:
It's important that variable names give us an idea of what the data actually is and/or what it will be used for. In this case, `replyUrlArray` or something similar seems more appropriate.